### PR TITLE
Alternative theme variant

### LINKED ISSUE #579: Theme variant logic
Pensar en la lógica del estado "inverse", tanto para fondos como para elementos que se posicionan encima de estos fondos.

- [x] [Analysis](https://www.figma.com/file/wyarSUpTJhKj2ij4VngxEL/A11y---Color-testing?node-id=38%3A4317)
- [x] #729
- [x] #730
- [x] https://github.com/Telefonica/mistica-design/issues/1185

Además será necesario abrir tickets para:

- [x] [Fix de las tabs sobre un inverse](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwD4AdAOwAI6kAlGMgBwhrO0xgBkAhgE8IAVwAu1eg0YBlcQLABrKTJlIAQhAAedNgKhRsNAOYBeYACYALAF8ptNWqQEBAIzKOnasjBIwwcRgoAEkaWG0LAAZbL28GX39A4LCIixMYcXkBIIAKKhAFN1SYbQKAGjoogEpY6Xi6TgBhAAsBUxgLXyyFPIKikrKQWrj4orILAG1gOiDtcXg6Atc3OgBGArpbSpm5haWQFbpLTe26XdL95fdj04BdOoaAelUnJCetbVfnI0w6MnEQn8FhmbkUShMaDE4UW4haMAAtjAAHSQEgQNBkZFg5SQ6FQDRodpQLb2UbOAjwpEANQEaGw7XEdGwZDCvExnX6aFEMAK33imh0egMRlMFhsZJo5IaLncnnq8USASCoXCpWijwa-z8ypSasiwAyPRyMHyhXcgwqVRGCu8zTaHS6mWyfXNxX1BRtWtmcqmF3mi2uqw2IC2O1ml0DhxuJ1DZ39V2jq1jWwe0oYL3T6g%2BOn5zielMRMFp9MZecYT1%2B5aQv3%2BgOBwFB4LxohhsypKLRGKxOIhUNbUAAgiQgmgaDkeDBSeWGILdPpDMZzFY7DOKXKswkdclVWlgDFN9qkirBulnb1Tf0LR6QJUapqGvb2hkncbXQMb16teM-RGAwcjhDMNzj-RMjhTeNQKjcD7gfJxM1tbNPmrSseH5d5smUV53hYdhOG4XhBBECQpHedAsFwQhiHIKRb0KDsyAQSYQA0EgBFEOiAFkIBwAE6TogB5SxBMsAB9Jo2LIbgwDoggdQAM04bAwAEOjqR4CAQDucoQDwbAoDhRj4EmB4gA) (Actualmente no usan text-preset y el divider no se adapta)

* Propuesta para soporte en el themeVariant de contextos alternative

## Elements
- Background `background`
- Background inverse `backgroundBrand`
- Background alternative `backgroundAlternative`
- Background inverse alternative `backgroundBrandSecondary`
- Normal item `backgroundContainer`
- Inverse item`backgroundBrand`
- Alternative item `backgroundAlternative`
- Alternative inverse item `backgroundBrandSecondary` (?)

## Combinations
- Background + Normal item
- Background + Alternative item
- Background + Inverse item
- Background + Alternative inverse item
- Background inverse + Normal item
- Background inverse + Alternative item
- Background inverse + Inverse item
- Background inverse + alternative inverse item
- Background alternative + Normal Item
- Background alternative + Alternative item
- Background alternative + Inverse item
- Background alternative + Alternative inverse item
- Background inverse alternative + Normal item
- Background inverse alternative + Alternative item
- Background inverse alternative + Inverse item
- Background inverse alternative + Alternative inverse item

## Important
Que funcione con dark mode también

## Actual example
https://ibit.ly/95sd
https://tinyurl.com/2xrbuw99

### LINKED ISSUE #1111: backgroundAlternative color in dark mode
## Description

Al permitir la posibilidad de que existan camas con `backgroundAlternative` se genera una serie de problemáticas en dark mode con todos aquellos componentes que usan `backgroundContainer`, ya que los dos pasan a usar `darkModeGrey`

[Ejemplo -> ](https://rebrand.ly/u422711)

La solución para este problema es cambiar el valor de `backgroundAlternative` a `darkModeBlack`.

**¿Qué ocurriría con aquellos componentes que utilizan el backgroundAlternative en su container?**

En el caso del callout, su variante inverse utiliza `backgroundContainer`, por lo que funcionaría sobre `backgroundAlternative` si siempre que cuando vaya en un contexto Alternative se pinte como `inverse`.

[Ejemplo -> ](https://rebrand.ly/1ntk1f0)

**¿Qué ocurriría con aquellos componentes que utilizan el backgroundAlternative en alguno de sus estados?**

En la propuesta actual ya se están perdiendo lo estados en darkMode:

[Ejemplo ->](https://rebrand.ly/g9q6dia)

En el caso de todos aquellos elementos que utilizan el backgroundAlternative para su estado :hover:

* Cards
* Listas
* PasswordField (ShowPassword)
* Select (OptionItem)

En estos casos esos estados deberían utilizar otro token de color que no sea el alternative:

Se añade complejidad a https://github.com/Telefonica/mistica-design/issues/677

Ya que estos estados deberían funcionar en los dos contextos, ya sea teniendo dos tokens diferentes o un token con un valor que no se iguale al background sobre el que están los elementos (en el caso del alternative)

### PR DESCRIPTION
**Pull request by aweell**
### New tokens added

In order to make the new alternative themeVariant work and avoid visual breaking changes in the current default and inverse, new tokens need to be added:

* `backgroundContainerBrand`
* `backgroundContainerAlternative`
* `neutralLowAlternative`

#### Callout

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `backgroundContainerAlternative` | `backgroundContainer` | `backgroundContainer `|

#### Tag type="inactive"

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `neutralLow` | `inverse` | `neutralLowAlternative`|

#### Chip unselected

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `backgroundContainer` | `backgroundContainer` | `backgroundAlternative`|

##### SnapCard isInverse

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `backgroundContainerBrand` | `brandHigh` | `backgroundContainerBrand`|

##### HighlightedCard isInverse

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `backgroundContainerBrand` | `brandHigh` | `backgroundContainerBrand`|

##### DispayDataCard isInverse

| Element | Default | Inverse | Alternative |
| :------ | :------- | :------- | :-------  |
| `background` | `backgroundContainerBrand` | `brandHigh` | `backgroundContainerBrand`|

### CODE REVIEW

**Review by yceballost — APPROVED** [review]
✨

**Comment by yceballost** [comment]
can you add vivo-new? 😇

File: tokens/blau.json [hunk 1]
```diff
@@ -25,6 +25,16 @@
       "type": "color",
       "description": "white"
     },
+    "backgroundContainerBrand": {
+      "value": "{palette.blauBluePrimary}",
+      "type": "color",
+      "description": "movistarBlue"
```
Comment (AnaMontes11) [inline tokens/blau.json]: Change it to "blauBluePrimary"

**Comment by aweell** [comment]
> can you add vivo-new? 😇

Done

File: tokens/blau.json (hunk 1 above)
Comment (yceballost) [inline tokens/blau.json]:
```suggestion
      "description": "blauBluePrimary"
```

**Review by yceballost — CHANGES_REQUESTED** [review]

**Review by AnaMontes11 — APPROVED** [review]

**Review by yceballost — APPROVED** [review]

**Pull request merged**